pkg/prometheus/promcfg_test.go

```diff
@@ -523,6 +749,7 @@ rule_files: []
 scrape_configs:
 - job_name: default/testservicemonitor1/0
   honor_labels: false
+  honor_timestamps: true
```

Comment:
Hmm do we want to always set this always? I understood only when the user explicitly sets it? cc @brancz

Reply:
Currently, the user doesn't have a way of setting `honor_timestamps` option and this way it is always enforced and clear what is happening. Also `true` is a default for prometheus.

Reply:
This seems right, under the premise that potentially we want to enable people to set this on a per servicemonitor/podmonitor basis.